### PR TITLE
Add support for ingesting from in-memory numpy arrays

### PR DESCRIPTION
**Pull request by NikolaosPapailiou**
This is implemented by writing the input numpy array in a TileDB array and using that for ingestion

### CODE REVIEW

File: apis/python/src/tiledb/vector_search/ingestion.py [hunk 1]
```diff
@@ -186,6 +193,51 @@ def read_source_metadata(
         else:
             raise ValueError(f"Not supported source_type {source_type}")
 
+    def write_input_vectors(
+        group: tiledb.Group,
+        input_vectors: np.array,
```
Comment (ihnorton) [inline apis/python/src/tiledb/vector_search/ingestion.py:198]: nit: `np.ndarray`

**Review by ihnorton — APPROVED** [review]
Some minor nits, LGTM and we can revisit the in-memory discussion later.

File: apis/python/test/common.py [hunk 2]
```diff
@@ -11,6 +11,36 @@ def xbin_mmap(fname, dtype):
     return np.memmap(fname, dtype=dtype, mode="r", offset=8, shape=(n, d))
 
 
+def get_vectors_fvec(file, dimensions=None, size=None):
```
[the remark this answers is not in the document]
Comment (NikolaosPapailiou) [reply apis/python/test/common.py]: Done

File: apis/python/src/tiledb/vector_search/ingestion.py (hunk 1 above)
Comment (NikolaosPapailiou) [reply apis/python/src/tiledb/vector_search/ingestion.py]: Done

File: apis/python/src/tiledb/vector_search/ingestion.py [hunk 3]
```diff
@@ -1407,9 +1461,22 @@ def consolidate_and_vacuum(
...
+            source_uri = write_input_vectors(
+                group=group,
+                input_vectors=input_vectors,
+                size=in_size,
+                dimensions=dimensions,
+                vector_type=vector_type,
+            )
+            source_type = "TILEDB_ARRAY"
+        else:
+            in_size, dimensions, vector_type = read_source_metadata(
```
[the remark this answers is not in the document]
Comment (NikolaosPapailiou) [reply apis/python/src/tiledb/vector_search/ingestion.py:1495]: Done

**Pull request merged**